### ISSUE TITLE
OpenGL shaders have too much/too little padding

### ISSUE THREAD

**Issue by Screwtapello** [byuu/bsnes#2]
When I have bsnes' video output set to "Scale" (as opposed to "Centre" or "Stretch"), I expect to see the video displayed in a 4:3 rectangle in the middle of the bsnes window. As of bsnes v107r4, this is not always the case.

With the [Pixellate](https://github.com/hizzlekizzle/quark-shaders/tree/master/Pixellate.shader) shader, the video output always covers the entire bsnes window, as though "Stretch" mode were active.

With the [Gameboy](https://github.com/hizzlekizzle/quark-shaders/tree/master/Gameboy.shader) shader, there is a black border between the inside edge of the bsnes window, and the area rendered by the shader. However, there is also another green border between the inside edge of the area rendered by the shader, and the actual console video output. I'm pretty sure the Gameboy shader is supposed to render the entire viewport green.

**Comment by byuu** [comment]
A bit out of my wheelhouse, but I think we're going to have to port the old quark pixel shaders into the new bsnes/higan style one by one. I had to change ruby to allow for black borders to allow OpenGL adaptive sync to work (it requires the OpenGL context span the entire monitor, and cannot be a child widget of a parent window anymore.)

**Comment by hizzlekizzle** [comment]
I think the problem is not the window itself, as making a simple shader that outputs vec4(texCoord.x) starts and stops in the right places. Instead, it seems there's some scaling happening after the shader is applied in fullscreen mode, and most shaders don't handle that very well. As mentioned in #49, masks fare particularly poorly.

The interlacing shader is a good, easy test, as it looks horrible with any post-scaling and/or non-integer scaling. It looks perfect in windowed mode, and it should look the same in fullscreen using the 'centered' output mode, but instead the lines are hideous and uneven.

**Comment by Screwtapello** [comment]
On the Discord, @hizzlekizzle poked at the Pixellate shader and discovered it had "width: 100%" and "height: 100%" in the output section, which would explain why its output gets stretched to cover the entire screen. It *doesn't* explain why Pixellate ever worked for older versions of bsnes.

After I learned that, I examined the Gameboy shader and discovered it lacked "width: 100%" and "height: 100%" output settings. Once I added them, it seemed to work properly too. Again, I don't understand how it ever worked properly.
